model/clusters_mgmt/v1/cloud_available_regions_resource.model

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright (c) 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Manages the collection of available regions of a cloud provider.
+resource AvailableRegions {
+	// Retrieves the list of regions of the cloud provider.
+	//
+	// IMPORTANT: This collection doesn't currently support paging or searching, so the returned
+	// `page` will always be 1 and `size` and `total` will always be the total number of regions
+	// of the provider.
+	method List {
```

Comment:
We should be using `method Post` for this resource, as we expect a body when requesting the list... Otherwise we would need to take in AWS credentials in the URL - not good :stuck_out_tongue:

Reply:
Based on https://github.com/openshift-online/ocm-api-metamodel/pull/95 we should use `method Search` here